### PR TITLE
[16.0] account_payment_purchase: replace api.onchange by computed field on purchase.order

### PR DESCRIPTION
**Pull request by alexis-via**
replace api.onchange by computed field on purchase.order
Add company_id in the domain of the fields of purchase.order
Add check_company=True on the 2 fields of purchase.order

### CODE REVIEW

File: account_payment_purchase/models/purchase_order.py [hunk 1]
```diff
@@ -7,19 +7,31 @@
...
     supplier_partner_bank_id = fields.Many2one(
         comodel_name="res.partner.bank",
+        compute="_compute_payment_mode",
+        readonly=False,
+        store=True,
+        precompute=True,
         string="Supplier Bank Account",
-        domain="[('partner_id', '=', partner_id)]",
+        domain="[('partner_id', '=', partner_id), "
+        "'|', ('company_id', '=', company_id), ('company_id', '=', False)]",
```
Comment (pedrobaeza) [inline account_payment_purchase/models/purchase_order.py]: You can simplify the domain this way:

`[("company_id", "in", [False, company_id])]`

File: account_payment_purchase/models/purchase_order.py [hunk 2]
```diff
@@ -7,19 +7,31 @@
 
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
+    _check_company_auto = True  # missing in the native purchase module !!!
```
Comment (pedrobaeza) [inline account_payment_purchase/models/purchase_order.py]: I think this side change shouldn't be done here.

File: account_payment_purchase/views/purchase_order_view.xml [hunk 3]
```diff
@@ -9,11 +9,7 @@
         <field name="inherit_id" ref="purchase.purchase_order_form" />
         <field name="arch" type="xml">
             <field name="payment_term_id" position="after">
-                <field
-                    name="payment_mode_id"
-                    domain="[('payment_type', '=', 'outbound')]"
-                    widget="selection"
-                />
+                <field name="payment_mode_id" widget="selection" />
```
Comment (pedrobaeza) [inline account_payment_purchase/views/purchase_order_view.xml]: AFAIK, this widget is conflicting in certain cases in v16. Better to use m2o

Comment (rvalyi) [reply account_payment_purchase/views/purchase_order_view.xml]: out of curiosity do you have some references or more details about the problems?

Comment (pedrobaeza) [reply account_payment_purchase/views/purchase_order_view.xml]: #1125

File: account_payment_purchase/models/purchase_order.py (hunk 1 above)
Comment (alexis-via) [reply account_payment_purchase/models/purchase_order.py]: done

File: account_payment_purchase/models/purchase_order.py (hunk 2 above)
Comment (alexis-via) [reply account_payment_purchase/models/purchase_order.py]: Of course ! It should be done in the purchase module !!!

File: account_payment_purchase/views/purchase_order_view.xml (hunk 3 above)
Comment (alexis-via) [reply account_payment_purchase/views/purchase_order_view.xml]: done

File: account_payment_purchase/models/purchase_order.py (hunk 2 above)
Comment (pedrobaeza) [reply account_payment_purchase/models/purchase_order.py]: Yes, but not being in core, propose it there, but do another module for this, not put this on a traversal module that has nothing to do with it.

**Review by luc-demeyer — APPROVED** [review]

**Comment by luc-demeyer** [comment]
Code review & testing ok.

I am wondering why precompute=True is set on the computed fields.
I tested by changing to precompute=False and the PR works in both cases.
@pedrobaeza 
do you think we need precompute=True in this case ?
When reading the doc in the ORM one would think it's required but it seems not to be the case.

**Comment by pedrobaeza** [comment]
It's because some fields are already precompute, and the dependency chain should be completely that way for avoiding double computing.

File: account_payment_purchase/models/purchase_order.py (hunk 2 above)
Comment (alexis-via) [reply account_payment_purchase/models/purchase_order.py]: done

**Comment by alexis-via** [comment]
I also converted (0, 0, {}) to Command.create([})
Ready to merge

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**